openshift_images/mongodb.adoc

```diff
@@ -10,61 +10,190 @@
+****
+
+== Testing
+
+This repository also provides a test framework which checks basic functionality of the MongoDB image.
 
-=== RHEL-7 Base Image
+Select whether the MongoDB image you are testing is RHEL-based or CentOS-based.
 
-----
-$ git clone https://github.com/openshift/mongodb.git
-$ cd mongodb
-$ make build TARGET=rhel7 VERSION=2.4
-----
+*RHEL-7 Base Image*
 
-=== CentOS-7 Base Image
+To test a RHEL-7-based MongoDB image, run the test on a properly subscribed RHEL machine.
 
-This image is also available on DockerHub. To download it use:
+. Navigate into the *_mongodb_* directory:
++
+****
+`$ cd mongodb`
+****
+. Test the image:
++
+****
+`$ make test TARGET=rhel7 VERSION=2.4`
+****
 
-----
-docker pull openshift/mongodb-24-centos7
-----
+*CentOS-7 Base Image*
 
-To build MongoDB image from scratch use:
+To test a CentOS-7-based MongoDB image:
 
-----
-$ git clone https://github.com/openshift/mongodb.git
-$ cd mongodb
-$ make build VERSION=2.4
-----
+. Navigate into the *_mongodb_* directory:
++
+****
+`$ cd mongodb`
+****
+. Test the image:
+****
+`$ make test VERSION=2.4`
+****
 
 [NOTE]
 ====
-If you omit the `VERSION` parameter, the build/test action is performed on all supported versions of MongoDB. Since version `2.4` is the only version supported, you can omit this parameter.
+If you omit the `VERSION` parameter, then the build/test action is performed on all supported versions of MongoDB. Since only version 2.4 is currently supported, you can omit this parameter.
 ====
 
+== Shared Volumes
 
-== Usage
+To set only mandatory required environment variables, and store the database in the shared *_/home/user/database_* directory on the host, execute the following command:
+
+****
+`docker run -d -e MONGODB_USERNAME=_<user>_ -e MONGODB_PASSWORD=_<password>_ -e MONGODB_DATABASE=_<database>_ -v /home/user/database:/var/lib/mongodb openshift/_<image>_`
+****
+
+=== Initializing Database
+
+The first time you use the shared volume and initialize the database, the
+database is created along with the database administrator user and the MongoDB
+root user (if you specify the `*MONGODB_ADMIN_PASSWORD*` environment variable).
+Afterwards, the MongoDB daemon starts up. If you are re-attaching the volume to
+another container, then the database user and the administrator user are not
+created, and the MongoDB daemon starts.
+
+== MongoDB Commands in Running Containers
+
+OpenShift uses https://www.softwarecollections.org/[Software Collections] to
+install and launch MongoDB. If you want to execute a MongoDB command
+inside of a running container (for debugging), you must prefix it with the `scl enable mongodb24` command.
+
+
+To execute MongoDB commands inside the container:
+
+****
+`$ scl enable mongodb24 -- mongo _<db_name>_ -u _<username>_ -p _<password>_`
+****
+
+To execute a command inside a running container from the host:
+
+****
+`$ docker exec -it _<CONTAINER_ID>_ scl enable mongodb24 /bin/bash`
```

Comment:
Now when I'm thinking about the logic here, it would be better to first how to enter the container, by:
`$ docker exec -it _<CONTAINER_ID>_  /bin/bash`

and then point out that the collection is enabled automatically, when entering the container, so the user can directly write mongodb specific commands:
`mongo _<db_name>_ -u _<username>_ -p _<password>_`

Not doing it vice verse - show how to run command and then show how to enter the container